orangecontrib/text/corpus.py

```diff
@@ -225,12 +230,23 @@ def from_file(cls, filename):
         table = Table.from_file(filename)
         return cls(table.X, table.Y, table.metas, table.domain, None)
 
+    def ngrams_iterator(self, join_with=None):
+        if join_with is None:
+            return (list(chain(*(nltk.ngrams(doc, n)
+                    for n in range(self.ngram_range[0], self.ngram_range[1]+1))))
+                    for doc in self.tokens)
+        else:
+            return (list(chain(*((join_with.join(ngram) for ngram in nltk.ngrams(doc, n))
+                    for n in range(self.ngram_range[0], self.ngram_range[1] + 1))))
+                    for doc in self.tokens)
```

Comment:
Looks nice. I suggest we set `join_with` to a space by default. Also it would be nice to have `.ngrams` property that would call `ngrams_iterator(' ')`.

So I suppose all other widgets should start using `corpus.ngrams` instead of `corpus.tokens`?